src/client/dfuse/pil4dfs/hook.c

```diff
@@ -172,77 +179,88 @@ read_map_file(char **buf, const char str_target[], char **pos)
 static void
 determine_lib_path(void)
 {
-	int   i, rc;
 	char *read_buff_map = NULL;
-	char *pPos = NULL, *pStart = NULL, *pEnd = NULL;
-	char  lib_ver_str[32], lib_dir_str[256];
+	char *pos, *start, *end;
+
+	read_map_file(&read_buff_map);
 
-	read_map_file(&read_buff_map, "/ld-2.", &pPos);
-	if (pPos == NULL) {
-		free(read_buff_map);
-		found_libc = 0;
+	pos = strstr(read_buff_map, "/lib64/ld-linux");
+	if (pos == NULL)
+		/* try a different format */
+		pos = strstr(read_buff_map, "/lib64/ld-2");
+	if (pos == NULL) {
 		printf("Warning: Failed to find ld.so.\n");
```

Comment:
Looking at the code for here it appears logging has been initialsed on all code-paths so there's no reason this function can't use our standard logging and allocation macros.

Reply:
Now daos logging functions are used.